### PR TITLE
decodeSensorData, decodeValveData, decodeDelayData, decodeMiscData

### PR DESCRIPTION
**Pull request by bshep**

### CODE REVIEW

File: messages/SLEquipmentConfigurationMessage.js [hunk 1]
```diff
@@ -117,4 +117,103 @@ exports.SLEquipmentConfigurationMessage = class SLEquipmentConfigurationMessage
...
+
+    this.sensors.poolSolarPresent = this.isBitSet(sensors[0], 1);
+    this.sensors.spaSolarPresent = this.isBitSet(sensors[0], 4);
+    this.sensors.thermaFloCoolPresent = this.isBitSet(sensors[1], 1);
+    this.sensors.solarHeatPumpPresent = this.isBitSet(sensors[2], 4);
+    this.sensors.thermaFloPresent = this.isBitSet(sensors[2], 5);
+  }
+
+  decodeValveData() {
+    var slaves = this.getSecondariesCount();
```
Comment (parnic) [inline messages/SLEquipmentConfigurationMessage.js:134]: I'm trying to stay away from the "slave" term.

File: messages/SLEquipmentConfigurationMessage.js [hunk 2]
```diff
@@ -117,4 +117,103 @@ exports.SLEquipmentConfigurationMessage = class SLEquipmentConfigurationMessage
...
+    this.misc = {};
+
+    if (this.isBitSet(this.miscDataArray[3], 0) === false) {
+      this.misc.intelliChem = false;
+    }
+
+    if (this.miscDataArray[4] !== 0) {
+      this.misc.spaManual = true;
+    } else {
+      this.misc.spaManuel = false;
```
Comment (parnic) [inline messages/SLEquipmentConfigurationMessage.js:212]: I assume this should be `spaManual` :)

File: messages/SLEquipmentConfigurationMessage.js [hunk 3]
```diff
@@ -117,4 +117,103 @@ exports.SLEquipmentConfigurationMessage = class SLEquipmentConfigurationMessage
...
+    }
+  }
+
+  isDualBody() {
```
Comment (parnic) [inline messages/SLEquipmentConfigurationMessage.js:216]: This function also exists in `SLControllerConfigMessage`. Perhaps another candidate for sharing, but I'm undecided on that.

File: messages/SLEquipmentConfigurationMessage.js (hunk 1 above)
Comment (bshep) [reply messages/SLEquipmentConfigurationMessage.js:134]: is 'secondaries' ok?

File: messages/SLEquipmentConfigurationMessage.js (hunk 3 above)
Comment (bshep) [reply messages/SLEquipmentConfigurationMessage.js:216]: i wouldn't share it, a generic SLMessage will not have the property 'controllerType' defined and will give an error, you could do some OOP magic but it may become hard to follow the code

Comment (parnic) [reply messages/SLEquipmentConfigurationMessage.js:216]: I was more thinking about a static method or something that both of these called. But duplicating is probably okay in this situation.
